backend/src/vehicles/vehicles.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CurrentUserService } from 'current-user/current-user.service';
+import { Repository } from 'typeorm';
+import { Vehicle } from './entities/vehicle.entity';
+
+interface VehicleArguments {
+  licensePlate?: string;
+  notes?: string;
+}
+
+@Injectable()
+export class VehicleService {
+  constructor(
+    @InjectRepository(Vehicle)
+    private vehiclesRepository: Repository<Vehicle>,
+    private currentUserService: CurrentUserService,
+  ) {}
+
+  async getAll(): Promise<Vehicle[]> {
+    const { organizationId } = this.currentUserService.getRequestCurrentUser();
+    return this.vehiclesRepository.find({
+      where: {
+        organization: { id: organizationId },
+      },
+      relations: {
+        organization: true,
+      },
+    });
+  }
+
+  async findOneById(id: number) {
+    const { organizationId } = this.currentUserService.getRequestCurrentUser();
+    return this.vehiclesRepository.findOne({
+      where: {
+        id,
+        organization: { id: organizationId },
+      },
+      relations: {
+        organization: true,
+      },
+    });
+  }
+
+  async checkIfExistsByLicensePlate(licensePlate: string) {
+    const { organizationId } = this.currentUserService.getRequestCurrentUser();
+    return (
+      (await this.vehiclesRepository.countBy({
+        licensePlate,
+        organization: { id: organizationId },
+      })) > 0
+    );
```

Comment:
Let's extract the call to the BD (so `this.vehiclesRepository.countBy(/* ... */)`) to a variable (we can call it `numberOfFoundVehicles`) and return only the comparison

```suggestion
    const numberOfFoundVehicles = await this.vehiclesRepository.countBy({
      licensePlate,
      organization: { id: organizationId },
    })

    return numberOfFoundVehicles > 0;
```